MonoGame.Framework/Utilities/Png/PngWriter.cs

```diff
@@ -0,0 +1,208 @@
+﻿// MonoGame - Copyright (C) The MonoGame Team
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using MonoGame.Utilities.ZLib;
+using Microsoft.Xna.Framework.Graphics;
+using Microsoft.Xna.Framework;
+
+namespace MonoGame.Utilities.Png
+{
+    public class PngWriter
+    {
+        private const int bitsPerSample = 8;
+        private ColorType colorType;
+        private Color[] textureData;
+
+        public PngWriter()
+        {
+            // TODO: currently ignoring alpha values while writing PNG images
+            colorType = ColorType.Rgb;
+        }
+
+        public Stream Write(Texture2D texture2D)
+        {
+            textureData = new Color[texture2D.Width * texture2D.Height];
+            texture2D.GetData<Color>(textureData);
```

Comment:
> Graphics.PackedVector classes are used to convert between pixel formats

Ahh...  that is true.  So in theory we would use that to cast from the hardware format to `Color`.  Something like...

```
var packedData = new Bgra4444[texture2D.Width * texture2D.Height]; 
texture2D.GetData<Bgra4444>(packedData);

var colorData = new Color[texture2D.Width * texture2D.Height];
for (var i=0; i < packedData.Length; i++)
   colorData[i] = new Color(packedData[i].ToVector4());
```
Obviously a bit more complex as we need to support all the different packed formats.... assuming here XNA does as well.